src/Features/VisualBasic/Portable/MakeMethodSynchronous/VisualBasicMakeMethodSynchronousCodeFixProvider.vb

```diff
@@ -0,0 +1,136 @@
+            Else
+                Dim newFunctionStatement = RemoveAsyncKeyword(functionStatement)
+                Return node.WithSubOrFunctionStatement(newFunctionStatement)
+            End If
+        End Function
+
+        Private Function FixSubBlock(node As MethodBlockSyntax) As SyntaxNode
+            Dim newSubStatement = RemoveAsyncKeyword(node.SubOrFunctionStatement)
+            Return node.WithSubOrFunctionStatement(newSubStatement)
+        End Function
+
+        Private Shared Function RemoveAsyncKeyword(subOrFunctionStatement As MethodStatementSyntax) As MethodStatementSyntax
+            Dim modifiers = subOrFunctionStatement.Modifiers
+            Dim asyncTokenIndex = modifiers.IndexOf(SyntaxKind.AsyncKeyword)
+
+            Dim newSubKeyword = subOrFunctionStatement.SubOrFunctionKeyword
+            Dim newModifiers As SyntaxTokenList
+            If asyncTokenIndex = 0 Then
+                ' Have to move the trivia on teh async token appropriately.
+                Dim asyncLeadingTrivia = modifiers(0).LeadingTrivia
+
+                If modifiers.Count > 1 Then
+                    ' Move the trivia to the next modifier;
+                    newModifiers = modifiers.Replace(
+                        modifiers(1),
+                        modifiers(1).WithPrependedLeadingTrivia(asyncLeadingTrivia))
+                    newModifiers = newModifiers.RemoveAt(0)
+                Else
+                    ' move it to the return type.
+                    newModifiers = modifiers.RemoveAt(0)
+                    newSubKeyword = newSubKeyword.WithPrependedLeadingTrivia(asyncLeadingTrivia)
+                End If
+            Else
+                newModifiers = modifiers.RemoveAt(asyncTokenIndex)
+            End If
+
+            Dim newSubStatement = subOrFunctionStatement.WithModifiers(newModifiers).WithSubOrFunctionKeyword(newSubKeyword)
+            Return newSubStatement
+        End Function
+
+        Private Function FixMultiLineLambdaExpression(node As MultiLineLambdaExpressionSyntax) As SyntaxNode
+            Dim header As LambdaHeaderSyntax = GetNewHeader(node)
+            Return node.WithSubOrFunctionHeader(header).WithLeadingTrivia(node.GetLeadingTrivia())
+        End Function
+
+        Private Function FixSingleLineLambdaExpression(node As SingleLineLambdaExpressionSyntax) As SingleLineLambdaExpressionSyntax
+            Dim header As LambdaHeaderSyntax = GetNewHeader(node)
+            Return node.WithSubOrFunctionHeader(header).WithLeadingTrivia(node.GetLeadingTrivia())
+        End Function
+
+        Private Shared Function GetNewHeader(node As LambdaExpressionSyntax) As LambdaHeaderSyntax
+            Dim header = DirectCast(node.SubOrFunctionHeader, LambdaHeaderSyntax)
+            Dim asyncKeywordIndex = header.Modifiers.IndexOf(SyntaxKind.AsyncKeyword)
+            Dim newHeader = header.WithModifiers(header.Modifiers.RemoveAt(asyncKeywordIndex))
```

Comment:
Why don't you do any of the trivia logic you do for `Sub/Function` statements here?